package/libpiksi/libpiksi/src/util.c

```diff
@@ -372,3 +383,122 @@ void setup_sigterm_handler(void (*handler)(int signum, siginfo_t *info, void *uc
     exit(-1);
   }
 }
+
+int setup_sigtimedwait(sigwait_params_t *params, int sig, time_t tv_sec)
+{
+  sigemptyset(&params->waitset);
+  sigaddset(&params->waitset, sig);
+  sigprocmask(SIG_BLOCK, &params->waitset, NULL);
+
+  update_sigtimedwait(params, tv_sec);
+
+  return 0;
+}
+
+int update_sigtimedwait(sigwait_params_t *params, time_t tv_sec)
+{
+  params->timeout.tv_sec = tv_sec;
+  params->timeout.tv_nsec = 0;
+
+  return 0;
+}
+
+int do_sigtimedwait(sigwait_params_t *params)
+{
+  int ret = sigtimedwait(&params->waitset, &params->info, &params->timeout);
+
+  if (-1 == ret && EAGAIN == errno) {
+    return 1;
+  } else {
+    return 0;
+  }
+}
+
+bool is_file(int fd)
+{
+  errno = 0;
+  int ret = lseek(fd, 0, SEEK_CUR);
+  return !(-1 == ret && errno == ESPIPE);
+}
+
+int run_with_stdin_file(const char *input_file,
```

Comment:
Let's add a few simple unit tests for this helper:
- Test overflow conditions
- Test no input file
- Test with an input file

Reply:
Any convenient example on the unit test framework or what did you have in mind? Need test some functions also...

Reply:
Seems you already found some unit testing examples, these projects also have unit tests:
- https://github.com/swift-nav/piksi_buildroot/tree/master/package/ports_daemon/ports_daemon/test
- https://github.com/swift-nav/piksi_buildroot/tree/master/package/sbp_fileio_daemon/sbp_fileio_daemon/test
- https://github.com/swift-nav/piksi_buildroot/tree/master/package/sbp_settings_daemon/sbp_settings_daemon/test

This commit is a good example of the transformation that typically happens when adding unit tests to a buildroot package:
- https://github.com/swift-nav/piksi_buildroot/commit/a5883f8dfa155468376c809b6da58391dcd35bea